onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -7041,20 +7041,22 @@ def aten_rsub_complex(self: TReal, other: TReal, alpha: float = 1.0) -> TReal:
     return aten_rsub(self, other, alpha)
 
 
-@torch_op("aten::scalar_tensor")
-def aten_scalar_tensor(s: float, dtype: int = FLOAT.dtype) -> TTensor:  # type: ignore[type-var]
+@torch_op("aten::scalar_tensor", trace_only=True)
+def aten_scalar_tensor(s: float, dtype: int = FLOAT.dtype) -> RealType:
     """scalar_tensor(Scalar s, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor"""
 
-    return op.Cast(s, to=dtype)
+    # Set trace_only=True because different if branches return different dtypes
+    # which is not supported in an ONNX function
+    return common_ops.cast_to(s, dtype=dtype)
 
 
-@torch_op("aten::scalar_tensor")
-def aten_scalar_tensor_sym_number(
-    s: Union[FLOAT, INT32, BOOL], dtype: int = FLOAT.dtype
-) -> TTensor:
+@torch_op("aten::scalar_tensor", trace_only=True)
```

Comment:
Have you tried it with dispatcher yet? I don't think this is supported... hmm.

Reply:
ONNX variants part

Reply:
I did. Which was not supported?

Reply:
I thought the dispatcher wouldn't be able to tell the two aten_scalar_tensor variants apart.

Reply:
It should be able to I think? Because the opschema is not affected in this change. One has attribute only and the other has one input and a single attribute

Reply:
I see